src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameProjection.java

```diff
@@ -283,50 +290,50 @@ private int nextExtendedPictureId()
 
         rewriteRtpInternal(rtpPacket);
 
-        if (originalSequenceNumber != vp8Frame.getStartingSequenceNumber())
+        int piggyBackUntilSequenceNumber
+            = vp8Frame.getMaxSequenceNumberSeenBeforeFirstPacket();
+
+        if (piggyBackUntilSequenceNumber < 0
```

Comment:
i'm a little confused by what's happening here.  we're making a new frame projection from the current one (what are the triggers for doing that, btw?) and we're signaling piggy-backed packets as part of creating that new frame projection?

Reply:
> i'm a little confused by what's happening here. we're making a new frame projection from the current one (what are the triggers for doing that, btw?) and we're signaling piggy-backed packets as part of creating that new frame projection?

A new frame has been accepted (hence, now it has to be projected). But right at this place we're not making a new frame projection, that's done in the makeNext method. Here we're merely projecting the RTP packets of the frame that refers to this frame projection.

Additionally, if this is the first packet of a frame, we piggyback any re-ordered packets that we've ignored.

Reply:
> A new frame has been accepted (hence, now it has to be projected).

this is because only packets which have been accepted (i.e. belong to frames which have been accepted) are passed to `rewriteRtp`?  What preconditions are true about a packet before it is passed to `rewriteRtp`?

> Here we're merely projecting the RTP packets of the frame that refers to this frame projection.
> 
> Additionally, if this is the first packet of a frame, we piggyback any re-ordered packets that we've ignored.

Ok, so the piggy-backing is used just in the case where we received the first packet of a frame and had already received some other packets so we return all of them at once.  If we later receive another old packet, we still return it I guess?  I'm thinking of a scenario where, say, a frame has 5 packets (seq num 1, 2, 3, 4, 5)

We receive packet 3 -> save it
We receive packet 4 -> save it
We receive packet 1 -> return packets 1, 3 and 4
We receive packet 2 -> we return it I assume?

Reply:
> What preconditions are true about a packet before it is passed to rewriteRtp?

There exists a frame projection (created in accept) that is able to project the packet; and the answer is yes to all the other questions.